src/test/ui/lint/lint-unused-mut-variables.rs

```diff
@@ -1,12 +1,40 @@
 // Exercise the unused_mut attribute in some positive and negative cases
 
-#![allow(unused_assignments)]
-#![allow(unused_variables)]
-#![allow(dead_code)]
 #![deny(unused_mut)]
+#![feature(param_attrs)]
+
+fn baz(
+    mut a: i32,
+    //~^ ERROR: variable does not need to be mutable
+    #[allow(unused_mut)] mut b: i32
```

Comment:
Could you also add a tests for a more complex pattern, like `(mut x, y): (i32, i32)`?